presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -269,6 +271,42 @@ public void checkCanDropColumn(ConnectorTransactionHandle transaction, Connector
         }
     }
 
+    @Override
+    public void checkCanDropBranch(ConnectorTransactionHandle transaction, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName)
+    {
+        MetastoreContext metastoreContext = new MetastoreContext(
+                identity, context.getQueryId().getId(),
+                context.getClientInfo(),
+                context.getClientTags(),
+                context.getSource(),
+                Optional.empty(),
+                false,
+                HiveColumnConverterProvider.DEFAULT_COLUMN_CONVERTER_PROVIDER,
+                context.getWarningCollector(),
+                context.getRuntimeStats());
+        if (!isTableOwner(transaction, identity, metastoreContext, tableName)) {
+            denyDropBranch(tableName.toString());
+        }
+    }
+
+    @Override
+    public void checkCanDropTag(ConnectorTransactionHandle transaction, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName)
```

Comment:
I wonder about the granularity of these methods. In other implementation (e.g. spark?) at what granularity to they enforce the ability to do CRUD operations on tags and branches? 

I'm thinking about a few cases

1. A group or user(s) can only access a certain set of branches or tags
2. A group or user(s) can only create branches starting from a specific branch
3. A group or user(s) can create tags

I know we're only implementing `DROP` but I want to understand the whole story for access control around branches and tags. Would we ever need to pass the branch/tag to these methods?

Reply:
I think the granularity of access control methods in Spark for CRUD operations on Iceberg tags and branches is limited by Spark's integration with external systems (such as file systems, catalogs, and security frameworks like Apache Ranger). 
For example, Ranger policies can define access controls at the table level, which could be extended to manage specific branches or tag-based access.
And like for cloud-based catalogs like AWS Glue, you can control access to Iceberg metadata (branches and tags) via IAM policies that grant or restrict specific operations.

Reply:
Thanks for the info. Would the parameters passed here as context have enough information for us to act at a similar level? I don't see anything in the method parameters that contains the branch name which I assume we would need to perform access control at a similar level.

Reply:
I would like to discuss around this, Systems like Ranger can define access controls at the table level, column level. So in this case I think handle the same access of drop branch & tags could be table based. As per I can think of branch & tag level policies then has to be maintained on engine side if we introduce that in here?

Reply:
@tdcmeehan What do you think about access control for branches and tags? Would be based on the parent table itself or based on the tags/branches?
My thinking was since there is no policies enforced based on branch/tags via security frameworks so this should owner the same policies as table?